### PR TITLE
feat(pkg): Add UUID utility package with comprehensive tests

### PR DESCRIPTION
**Pull request by captain-corgi**
## Description
This PR adds a comprehensive UUID utility package with full test coverage using Table Driven Testing patterns.

## Type of Change
- [x] New feature
- [x] Documentation update

## Module
- [x] Other: Shared utilities (pkg/uuid)

### ✨ New Features
- **UUID Package**: Added `pkg/uuid` with Google UUID wrapper functions
  - `NewGoogle()`: Generate new UUID v4
  - `ParseGoogle(string)`: Parse UUID string with error handling
  - `MustParseGoogle(string)`: Parse UUID string with panic on error

### 🧪 Testing
- **Comprehensive Unit Tests**: Added `google_test.go` with Table Driven Testing
  - ✅ 100% code coverage
  - ✅ 21 test cases covering valid/invalid inputs
  - ✅ Panic testing for `MustParseGoogle`
  - ✅ Performance benchmarks for all functions
  - ✅ Edge case validation (empty strings, invalid formats, etc.)

### 📊 Test Results
```
BenchmarkNewGoogle-8         4,092,978    296.6 ns/op
BenchmarkParseGoogle-8      10,364,223    100.8 ns/op  
BenchmarkMustParseGoogle-8  11,814,889     99.32 ns/op
```

### 🏗️ Project Structure
- Updated Go workspace configuration
- Added proper module dependencies
- Follows Clean Architecture guidelines

## Testing
- [x] Unit tests added/updated
- [x] Manual testing completed
- [x] All tests pass with 100% coverage
- [x] Benchmark tests included

## Documentation
- [x] Code properly documented
- [x] Tests follow Table Driven Testing pattern
- [x] Follows Go best practices

## Checklist
- [x] Code follows project conventions
- [x] Self-review completed
- [x] Tests pass locally
- [x] No merge conflicts
- [x] Follows Conventional Commits format

## Next Steps
This UUID package provides the foundation for:
- Event-driven architecture implementations
- Unique identifier generation across microservices
- Domain entity identification in DDD patterns

---
Pull Request opened by [Augment Code](https://www.augmentcode.com/) with guidance from the PR author

## Summary by CodeRabbit

- **New Features**
  - Introduced new utilities for generating and parsing UUIDs, providing easy-to-use functions for working with UUID strings.

- **Bug Fixes**
  - None.

- **Documentation**
  - Updated project documentation to reflect directory structure changes.
  - Removed outdated documentation for shared components.

- **Tests**
  - Added comprehensive tests and performance benchmarks for the new UUID utility functions.

- **Chores**
  - Added Go module configuration for the new utilities package.
  - Removed legacy setup scripts and obsolete shared documentation.

### CODE REVIEW

File: pkg/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module github.com/captain-corgi/learning-event-driven/pkg
+
+go 1.24.0
+
+require github.com/google/uuid v1.6.0
```
[the remark this answers is not in the document]
Comment (captain-corgi) [reply pkg/go.mod:5]: This is latest version already, close this comment
